src/containers/survey/HourlyAppUsageSurvey.js

```diff
@@ -0,0 +1,241 @@
+      }
+      else {
+        otherChildHourlySelections.update(appName, Set(), (current) => current.add(id));
+      }
+      return {
+        ...state,
+        otherChildHourlySelections
+      };
+    }
+
+    case ACTIONS.TOGGLE_INSTRUCTIONS_MODAL: {
+      const { visible } = action;
+      return {
+        ...state,
+        isInstructionsModalVisible: visible
+      };
+    }
+
+    case ACTIONS.CHILD_SELECT_TIME: {
+      const { appName, id } = action;
+      const { childHourlySelections } = state;
+
+      if (childHourlySelections.get(appName, Set()).has(id)) {
+        childHourlySelections.update(appName, Set(), (current) => current.delete(id));
+      }
+      else {
+        childHourlySelections.update(appName, Set(), (current) => current.add(id));
+      }
+      return {
+        ...state,
+        childHourlySelections
+      };
+    }
+    case ACTIONS.NEXT_STEP: {
+      return {
+        ...state,
+        step: state.step + 1
+      };
+    }
+    case ACTIONS.PREV_STEP: {
+      return {
+        ...state,
+        step: state.step - 1
+      };
+    }
+    case ACTIONS.CONFIRM_SUBMIT: {
+      return {
+        ...state,
+        isSubmissionConfirmed: true,
+        isConfirmModalVisible: false
+      };
+    }
+    case ACTIONS.CANCEL_SUBMIT: {
+      return {
+        ...state,
+        isConfirmModalVisible: false
+      };
+    }
+    case ACTIONS.SHOW_CONFIRM_MODAL: {
+      return {
+        ...state,
+        isConfirmModalVisible: true
+      };
+    }
+    default:
+      return state;
+  }
+};
+
+type Props = {
+  data :Map;
+  date :string;
+  submitSurveyRS :?RequestState;
+  participantId :string;
+  organizationId :UUID;
+  studyId :UUID ;
+};
+
+const HourlyAppUsageSurvey = (props :Props) => {
+  const {
+    data,
+    date,
+    studyId,
+    organizationId,
+    participantId,
+    submitSurveyRS,
+  } = props;
+
+  const storeDispatch = useDispatch();
+
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  const {
+    step,
+    childOnlyApps,
+    isConfirmModalVisible,
+    childHourlySelections,
+    otherChildHourlySelections,
+    isSubmissionConfirmed,
+    isInstructionsModalVisible
+  } = state;
+
+  const createSubmissionData = () => {
+
+    const childOnlyIds = childOnlyApps
+      .map((app) => data.getIn([app, 'entities']).map((entity) => entity.keySeq())).flatten();
+
+    const otherIds = childHourlySelections.valueSeq()
+      .concat(otherChildHourlySelections.valueSeq()).toSet().flatten();
+
+    return childOnlyIds.concat(otherIds)
+      .toMap()
+      .mapEntries((entry) => [entry[0], fromJS({ [USER_FQN.toString()]: ['Target child'] })])
+      .toJS();
+  };
+
+  useEffect(() => {
+    if (isSubmissionConfirmed) {
+      storeDispatch(submitSurvey({
+        submissionData: createSubmissionData(),
+        organizationId,
+        participantId,
+        studyId
+      }));
+    }
+  }, [isSubmissionConfirmed]);
```

Comment:
Please do not ignore the React Hook useEffect dependency warning. Looks like you can just place createSubmissionData logic inside the hook.

https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies